jinahub/encoders/image/CLIPImageEncoder/clip_image.py

```diff
@@ -81,35 +80,36 @@ def encode(self, docs: Optional[DocumentArray], parameters: dict, **kwargs):
             The accepted keys are ``traversal_paths`` and ``batch_size`` - in their
             absence their corresponding default values are used.
         """
-        if docs:
-            document_batches_generator = get_docs_batch_generator(
-                docs,
-                traversal_path=parameters.get(
-                    "traversal_paths", self.default_traversal_paths
-                ),
-                batch_size=parameters.get("batch_size", self.default_batch_size),
-                needs_attr="blob",
-            )
+        if docs is None:
+            return
 
-            with torch.no_grad():
-                for batch_docs in document_batches_generator:
-                    blob_batch = [d.blob for d in batch_docs]
-                    if self.use_default_preprocessing:
-                        tensor = self._generate_input_features(blob_batch.copy())
-                    else:
-                        tensor = {
-                            "pixel_values": torch.tensor(
-                                blob_batch.copy(),
-                                dtype=torch.float32,
-                                device=self.device,
-                            )
-                        }
+        traversal_paths = parameters.get("traversal_paths", self.traversal_paths)
+        batch_size = parameters.get("batch_size", self.batch_size)
+        document_batches_generator = docs.batch(
+            traversal_paths=traversal_paths,
+            batch_size=batch_size,
+            require_attr="blob",
+        )
+
+        with torch.inference_mode():
+            for batch_docs in document_batches_generator:
+                blob_batch = [d.blob for d in batch_docs]
+                if self.use_default_preprocessing:
+                    tensor = self._generate_input_features(blob_batch.copy())
+                else:
+                    tensor = {
+                        "pixel_values": torch.tensor(
+                            blob_batch.copy(),
+                            dtype=torch.float32,
+                            device=self.device,
+                        )
+                    }
 
-                    embeddings = self.model.get_image_features(**tensor)
-                    embeddings = embeddings.cpu().numpy()
+                embeddings = self.model.get_image_features(**tensor)
+                embeddings = embeddings.cpu().numpy()
 
-                    for doc, embed in zip(batch_docs, embeddings):
-                        doc.embedding = embed
+                for doc, embed in zip(batch_docs, embeddings):
+                    doc.embedding = embed
```

Comment:
i see `batch` function create a generator of document arrays, so each of `batch_docs` should be a `DocumentArray`, then we can use the setter directly without looping? should be faster:

```python
da.embeddings = embeddings
```

it also applies to other loops

Reply:
Thanks, good observation

Reply:
@bwanglzu it seems that this causes some issues - there might be something wrong with how we implement getters and setters in the core, I will investigate. For now I will revert my last commit, and try to add this feature back once I finish investigating the problem in another PR.

Reply:
sounds good to me!